### ISSUE TITLE
`dataclass-field-order` incorrectly counts inherited `ClassVar` as a defaulted field on Python 3.14

### ISSUE THREAD

**Issue by juanzolotoochin** [astral-sh/ty#4265]
### Summary

Under Python 3.14, ty reports `dataclass-field-order` when a dataclass inherits from another dataclass containing a defaulted `ClassVar`.

The `ClassVar` should not be treated as a dataclass field. The equivalent code constructs successfully at runtime.

## Repro

Using ty 0.0.71 and `marshmallow-dataclass`, create these files:

`base.py`:

```python
from typing import ClassVar

from marshmallow_dataclass import dataclass


@dataclass
class EventRoot:
    Schema: ClassVar[type[object]] = object


@dataclass
class RequiredEventFields(EventRoot):
    organization_id: str
```

`base.pyi`:

```python
from typing import ClassVar

from marshmallow_dataclass import dataclass


@dataclass
class EventRoot:
    Schema: ClassVar[type[object]]


@dataclass
class RequiredEventFields(EventRoot):
    organization_id: str
```

`example.py`:

```python
from marshmallow_dataclass import dataclass

from base import RequiredEventFields


@dataclass
class AuditEvent(RequiredEventFields):
    event_name: str


event = AuditEvent(organization_id="org", event_name="login")
assert event.organization_id == "org"
assert event.event_name == "login"
```

Run:

```bash
uv run --with ty==0.0.71 --with marshmallow-dataclass \
  ty check --python-version 3.14 example.py
```

## Actual result

```text
example.py:8:5: error: dataclass-field-order:
Required field `event_name` cannot be defined after fields with default values
```

## Expected result

No `dataclass-field-order` diagnostic.

`Schema` is annotated as a `ClassVar`, so it is excluded from dataclass fields and should not participate in field-order validation. The example also executes successfully under Python 3.14.

## Additional context

This occurs when the inherited dataclass is consumed through a `.pyi` stub. The stub preserves the `ClassVar` annotation, but ty appears to treat it as a defaulted instance field while validating the subclass.

**Comment by carljm** [comment]
Thanks for the report!

This is a real bug, but `ClassVar` is a red herring. We do correctly exclude `ClassVar` from consideration as a dataclass field. The actual problem is the `organization_id` field -- because it is in a stub, and we treat all declarations as also bindings in stubs, we wrongly consider this field to have a default when it does not.
